### PR TITLE
Updating product quant in shopcarts PUT 

### PR DESCRIPTION
**Pull request by carpetri**
PUT On /shopcarts/uid/products/pid

All tests in place.
Coverage of 98%

Added utility function from rofrano's code to check for media type.

### CODE REVIEW

File: models/shopcart.py [hunk 1]
```diff
@@ -53,13 +53,19 @@ def deserialize(self,data):
                 if type( data['products'] ) == dict:
                     # ** JSON.DUMP MAKES KEYS TO STR
                     prods = { int(p):int(q) for (p,q) in data['products'].items() }
+                    for pid,quant in prods.items():
+                        if quant == 0:
+                            self.delete_product(pid)
+                            prods.pop(pid,None)
                 else:
```
Comment (radhikamattoo) [inline models/shopcart.py]: I'm getting the `Updating a product to 0` test failing at this part:
`for pid,quant in prods.items():` 
with the error:
`RuntimeError: dictionary changed size during iteration`

Comment (carpetri) [reply models/shopcart.py]: I'll check it! thanks

Comment (carpetri) [reply models/shopcart.py]: I don't know how i missed that. Thanks @radhikamattoo

Comment (carpetri) [reply models/shopcart.py]: I don't know how i missed that. Thanks @radhikamattoo

**Comment by thoughts1053** [comment]
@carpetri when you get a chance can you fix conflicts? I have a branch almost ready for a PR but I want to get your changes in first as there's some overlap.

**Comment by carpetri** [comment]
done! @thoughts1053

**Pull request merged**